pkg/payment/refund/client_test.go

```diff
@@ -0,0 +1,469 @@
+package refund
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/mercadopago/sdk-go/pkg/config"
+	"github.com/mercadopago/sdk-go/pkg/internal/httpclient"
+)
+
+var (
+	responseJSON, _ = os.Open("../../../resources/mocks/payment/refund/response.json")
+	response, _     = io.ReadAll(responseJSON)
+
+	listResponseJSON, _ = os.Open("../../../resources/mocks/payment/refund/list_response.json")
+	listResponse, _     = io.ReadAll(listResponseJSON)
+)
+
+func TestCreate(t *testing.T) {
+	type fields struct {
+		config *config.Config
+	}
+	type args struct {
+		ctx context.Context
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Response
+		wantErr string
+	}{
+		{
+			name: "should_fail_to_create_request",
+			fields: fields{
+				config: nil,
+			},
+			args: args{
+				ctx: nil,
+			},
+			want:    nil,
+			wantErr: "error creating request: net/http: nil Context",
+		},
+		{
+			name: "should_fail_to_send_request",
+			fields: fields{
+				config: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							return nil, fmt.Errorf("some error")
+						},
+					},
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+			},
+			want:    nil,
+			wantErr: "transport level error: some error",
+		},
+		{
+			name: "should_fail_to_unmarshaling_response",
+			fields: fields{
+				config: &config.Config{
+					Requester: &httpclient.Mock{
+						DoMock: func(req *http.Request) (*http.Response, error) {
+							stringReader := strings.NewReader("invalid json")
+							stringReadCloser := io.NopCloser(stringReader)
+							return &http.Response{
+								Body: stringReadCloser,
+							}, nil
+						},
+					},
+				},
+			},
+			args: args{
+				ctx: context.Background(),
+			},
+			want:    nil,
+			wantErr: "invalid character 'i' looking for beginning of value",
+		},
+		{
+			name: "should_return_formatted_response",
```

Comment:
se resposta é resposta, mudar o nome do teste
```suggestion
			name: "should_return_response",
```